avm/res/storage/storage-account/file-service/main.bicep

```diff
@@ -18,7 +18,11 @@ param shareDeleteRetentionPolicy object = {
   days: 7
 }
 
+@description('Optional. Specifies CORS rules for the File service. You can include up to five CorsRule elements in the request. If no CorsRule elements are included in the request body, all CORS rules will be deleted, and CORS will be disabled for the File service.')
+param corsRules array = []
```

Comment:
Maybe it would be good to offer a User-defined-type here, but regardless, did you try making this nullable:
```bicep
param corsRules object[]?

(...)
    cors: !empty(corsRules) ? {
      corsRules: corsRules
    } : null
(...)
```
Maybe this will even preserve the current `corsRules`